RepositoryCompiler/CodeModel/CodeParsers/CSharp/CaDETMemberMetricCalculator.cs

```diff
@@ -121,5 +137,404 @@ private int CountOccurrences(string text, string pattern)
+        {
+            return method.DescendantNodes().OfType<ExpressionSyntax>()
+                .Count(n => n.IsKind(SyntaxKind.ParenthesizedExpression));
+        }
+
+        private int CountNumberOfLambdaExpressions(MemberDeclarationSyntax method)
+        {
+            return method.DescendantNodes().OfType<LambdaExpressionSyntax>().Count();
+        }
+
+        private int CountMaxNestedBlocks(MemberDeclarationSyntax method)
+        {
+            if (!method.Kind().Equals(SyntaxKind.MethodDeclaration))
+            {
+                return 0;
+            }
+            BaseMethodDeclarationSyntax baseMethod = (BaseMethodDeclarationSyntax)method;
+            var blocks = baseMethod.Body.DescendantNodes().OfType<BlockSyntax>().ToList();
+
+            List<int> blocksAncestors = new List<int>();
+            foreach (var block in blocks)
+            {
+                blocksAncestors.Add(block.Ancestors().Count(a => a.IsKind(SyntaxKind.Block)));
+            }
+
+            if (!blocksAncestors.Any())
+            {
+                return 0;
+            }
+            return blocksAncestors.Max();
+        }
+
+        private int CountNumberOfUniqueWords(MemberDeclarationSyntax method)
+        {
+            if (!method.Kind().Equals(SyntaxKind.MethodDeclaration))
+            {
+                return 0;
+            }
+            BaseMethodDeclarationSyntax baseMethod = (BaseMethodDeclarationSyntax)method;
+            string methodBody = baseMethod.Body.ToString();
+            List<string> lines = GetLinesWithoutComments(methodBody);
+            List<string> words = GetWordsFromLines(lines);
+            words = FilterWords(words);
+            List<string> individualWords = BreakWords(words);
+            individualWords = FindAdditionalWords(individualWords);
+            return CleanWordsFromSymbols(individualWords).Distinct().ToList().Count();
+        }
+
+        private List<string> GetLinesWithoutComments(string methodBody)
+        {
+            List<string> lines = methodBody.Split("\n")
+                .Select(line => line.Trim())
+                .Where(line => !line.StartsWith("//"))
+                .ToList();
+            return RemoveMultiLineComments(lines);
+        }
+
+        private List<string> RemoveMultiLineComments(List<string> lines)
+        {
+            List<string> linesToRemove = new List<string>();
+            for (int i = 0; i < lines.Count(); i++)
+            {
+                if (lines[i].StartsWith("/*"))
+                {
+                    linesToRemove.Add(lines[i]);
+                    int j = 1;
+                    while (!lines[i + j].EndsWith("*/"))
+                    {
+                        linesToRemove.Add(lines[i + j]);
+                        j++;
+                    }
+                    linesToRemove.Add(lines[i + j]);
+                }
+            }
+            lines.RemoveAll(line => linesToRemove.Contains(line));
+            return lines;
```

Comment:
The code to solve this problem is located a few hundred lines above: https://github.com/Clean-CaDET/platform/blob/master/RepositoryCompiler/CodeModel/CodeParsers/CSharp/CaDETMemberMetricCalculator.cs#L31-L37 which you can extract into a private method that will be used by both calculations.